src/libraries/Microsoft.Extensions.Logging/src/LoggerFactory.cs

```diff
@@ -211,12 +212,19 @@ private void AddProviderRegistration(ILoggerProvider provider, bool dispose)
 
         private LoggerInformation[] CreateLoggers(string categoryName)
         {
-            var loggers = new LoggerInformation[_providerRegistrations.Count];
+            var loggers = new List<LoggerInformation>(_providerRegistrations.Count);
             for (int i = 0; i < _providerRegistrations.Count; i++)
             {
-                loggers[i] = new LoggerInformation(_providerRegistrations[i].Provider, categoryName);
+                var loggerInformation = new LoggerInformation(_providerRegistrations[i].Provider, categoryName);
+
+                // We do not need to check for NullLogger<T>.Instance as no provider would reasonably return it (the <T> handling is at
+                // outer loggers level, not inner level loggers in Logger/LoggerProvider).
+                if (loggerInformation.Logger != NullLogger.Instance)
+                {
+                    loggers.Add(loggerInformation);
+                }
             }
-            return loggers;
+            return [..loggers];
```

Comment:
>return [..loggers]; [](http://example.com/codeflow?start=12&length=19)

Why you need to use Range syntax here? Please use `ToArray` directly.

Reply:
I don't. I think roslyn was bugging me it could be simplified (I disagree this simpler but alas). Will revert :)